network/test/cohort1/network_test.go

```diff
@@ -469,10 +479,14 @@ func (suite *NetworkTestSuite) TestUnicastRateLimit_Bandwidth() {
 	unittest.RequireComponentsReadyBefore(suite.T(), 1*time.Second, newNet)
 
 	// registers an engine on the new network so that it can receive messages on the TestNetworkChannel
-	newEngine := &mocknetwork.MessageProcessor{}
+	newEngine := mocknetwork.NewMessageProcessor(suite.T())
 	_, err = newNet.Register(channels.TestNetworkChannel, newEngine)
 	require.NoError(suite.T(), err)
-	newEngine.On("Process", channels.TestNetworkChannel, suite.ids[0].NodeID, mockery.Anything).Return(nil)
+
+	callCount := 0
```

Comment:
is it safe to use a non-atomic here? Do we have a guarantee that everything runs in the same goroutine?